### ISSUE TITLE
Test results depend on ground truth mesh

### ISSUE THREAD

**Issue by Silvarebacca** [YujiaLiu76/point2building#5]
Dear Author,

When I tried to test the model with the input point cloud, I found that the ground truth mesh information is used in the provided code. If I delete all the related information, the results become much worse. Could you please tell me how to handle this problem? 

Thank you very much.

**Comment by ShuaibZyx** [comment]
How did you do it? How do I select files that match the test set and the training set? @Silvarebacca

**Comment by zhuangzhi66** [comment]
> Dear Author,
> 
> When I tried to test the model with the input point cloud, I found that the ground truth mesh information is used in the provided code. If I delete all the related information, the results become much worse. Could you please tell me how to handle this problem?
> 
> Thank you very much.

Have you solved it yet

**Comment by philferriere** [comment]
There seems to be significant data leakage, indeed...

I found two places where input point clouds are preprocessed using *GT* vertex information that is not available at inference time.

The first is in "hallucinating" ground points from some min vertex z:

```python
minz = -np.max(vertices[:,2])
added_pts = pts.copy()
added_pts[:,2] = minz
pts = np.vstack((pts, added_pts))
```

This effectively doubles the number of points in the input pointcloud, adding a ground projection/shadow of the points on a plane defined by `minz`. Even the approach used to compute `minz` is a bit unexpected: why isn't that simply `np.min(vertices[:,2])`? More importantly, you cannot perform this operation at inference time at all since the vertices are what you are effectively trying to predict with the first stage...

The second is in scaling the pointcloud values, again using vertex information:

```python
vertices, vertices_scale = data_utils.normalize_vertices_scale(vertices, return_scale=True)
pts = pts / vertices_scale
```

Here too, this operation cannot be done at inference time...

@Silvarebacca, 

1/ Were there other places where you found data leakage?

2/ Was your attempt at a workaround simply to not scale and not add the "shadow" points? Or, perhaps, use the minz of the pointcloud itself instead (which I do realize can be prolematic if, for instance, all the points you have in the pointcloud come from a roof structure and no ground points are present at all) for the added points.

3/ Has anyone found a legitimate way to make the approach in this paper work?

Best regards, -- Phil

**Comment by eyegotthis** [comment]
@Silvarebacca @philferriere interesting observation! I was about to test it out on a custom dataset but now a bit hesitant towards it. Did you find a work-around on the data leakage? Were you able to produce promising results? looking forward

Cheers
Esas
